docs/rtk-query/usage-with-typescript.mdx

```diff
@@ -398,6 +398,7 @@ When using the function notation, both the `providesTags` and `invalidatesTags`
 - result: `ResultType` | `undefined` - The result returned by a successful query. The type corresponds with `ResultType` as [supplied to the built endpoint](#typing-query-and-mutation-endpoints). In the error case for a query, this will be `undefined`.
 - error: `ErrorType` | `undefined` - The error returned by an errored query. The type corresponds with `Error` as [supplied to the `baseQuery` for the api](#typing-a-basequery). In the success case for a query, this will be `undefined`.
 - arg: `QueryArg` - The argument supplied to the `query` property when the query itself is called. The type corresponds with `QueryArg` as [supplied to the built endpoint](#typing-query-and-mutation-endpoints).
+- meta: `MetaType` | `undefined` - The optional `meta` value returned by `baseQuery`
```

Comment:
Hmm, this is tricky to communicate. When using `fetchBaseQuery`, this will actually be a combination of `baseQueryMeta` (the type `FetchBaseQueryMeta`) and a _maybe_ user-defined `SomeCustomInjectedMetaAUserProvides`. This results in an object that looks like: `{ request: Request, response?: Response, ...someCustomInjectedMeta }`. This will be present for any `fulfilled` or `rejected` request.

This might be fine for now, but we'll definitely want to revisit this as well as the `Adding Meta information to queries` section in `customizing-queries.md`

Reply:
I appreciate the clarification. Let me chew on this feedback and I can push another iteration.

Reply:
little confused by this comment - when is baseQueryMeta *not* determined by the baseQuery? (#4098 not withstanding, which i just discovered)